Samarth/cfm

AxiomSamarth opened the pull request:
This PR includes the interface for HPE CFM to login, logout and to fetch the fabric details from the HPE CFM instance.

Commit:
Added cfm cli to hpe cli

This commit includes the command line interface to communicate with and operate on HPE Composable Fabric.

bsoutham committed:
work on tests.  now ~83% coverage

AxiomSamarth committed:
Merge pull request #1 from bsoutham/bsoutham/tests

bsoutham/tests

Commit:
Added the 'get fabrics' for cfm interface

This commit also removes some custom logging features which were added before to the connectapi pkg

Commit:
Merge remote-tracking branch 'upstream/master' into samarth/cfm

Commit:
Update integrations.go

Fixes of some refactoring issue.

bsoutham reviewed (changes requested):
everything under cfm/ seems to be the CFM SDK?
Is there a different repo that contains the CFM SDK?  Published somewhere else on github.   In the CLI project isn't the right place to have the CFM SDK content.  It should just be added as a dependency of the CLI.

Havnen't reviewed code beyond that.. once we get the SDK placement issue resolved - we can continue the review.

bsoutham commented:
I have enabled builds on PR from forks.  So next submit this should actually do a build and run checks.

AxiomSamarth commented:
> everything under cfm/ seems to be the CFM SDK?
> Is there a different repo that contains the CFM SDK? Published somewhere else on github. In the CLI project isn't the right place to have the CFM SDK content. It should just be added as a dependency of the CLI.
> 
> Havnen't reviewed code beyond that.. once we get the SDK placement issue resolved - we can continue the review.

Yes, Blaine. Everything under cfm/ is the CFM SDK. No, it is not published elsewhere on GitHub. I can host it as a separate repo and add it as a dependency. Will do it.

Commit:
Removed the dependencies of CFM SDK

This commit removes the SDK content from the CLI codebase and treats it only as a dependency.

AxiomSamarth commented:
> everything under cfm/ seems to be the CFM SDK?
> Is there a different repo that contains the CFM SDK? Published somewhere else on github. In the CLI project isn't the right place to have the CFM SDK content. It should just be added as a dependency of the CLI.
> 
> Havnen't reviewed code beyond that.. once we get the SDK placement issue resolved - we can continue the review.

Hi Blaine,

I have created a separate repo https://github.com/AxiomSamarth/cfm which has the CFM SDK and in the CLI Project it is now treated only as a dependency.
